ecs-init/docker/docker_test.go

```diff
@@ -985,3 +987,99 @@ func TestDefaultIsPathValid(t *testing.T) {
+		},
+		{
+			name:                                 "Credentials Fetcher from env, no Credentials Fetcher from config file",
+			credentialsFetcherHostFromEnv:        "unix:///var/credentials-fetcher/socket/credentials_fetcher.sock",
+			credentialsFetcherHostFromConfigFile: "",
+			expectedBind:                         "/var/credentials-fetcher/socket/credentials_fetcher.sock:/var/credentials-fetcher/socket/credentials_fetcher.sock",
+		},
+		{
+			name:                                 "Credentials Fetcher from env, invalid Credentials Fetcher from config file",
+			credentialsFetcherHostFromEnv:        "unix:///var/credentials-fetcher/socket/credentials_fetcher.sock",
+			credentialsFetcherHostFromConfigFile: "invalid",
+			expectedBind:                         "/var/credentials-fetcher/socket/credentials_fetcher.sock:/var/credentials-fetcher/socket/credentials_fetcher.sock",
+		},
+		{
+			name:                                 "Credentials Fetcher host from env, Credentials Fetcher from config file",
+			credentialsFetcherHostFromEnv:        "unix:///var/credentials-fetcher/socket/credentials_fetcher.sock.1",
+			credentialsFetcherHostFromConfigFile: "unix:///var/credentials-fetcher/socket/credentials_fetcher.sock.1",
+			expectedBind:                         "/var/credentials-fetcher/socket/credentials_fetcher.sock.1:/var/credentials-fetcher/socket/credentials_fetcher.sock.1",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			os.Setenv("CREDENTIALS_FETCHER_HOST", tc.credentialsFetcherHostFromEnv)
+			defer os.Unsetenv("CREDENTIALS_FETCHER_HOST")
+
+			bind, _ := getCredentialsFetcherSocketBind()
+			assert.Equal(t, tc.expectedBind, bind)
+		})
+	}
+}
+
+func TestInstanceDomainJoined(t *testing.T) {
+	execCommand = fakeExecCommand
+	defer func() { execCommand = exec.Command }()
+
+	defer func() { execLookPath = exec.LookPath }()
+
+	execLookPath = func(name string) (string, error) {
+		return "/usr/sbin/realm", nil
+	}
+
+	out := isDomainJoined()
+	assert.True(t, out)
+}
+
+func TestInstanceDomainJoinedRealmNotFound(t *testing.T) {
+	execCommand = fakeExecCommand
+	defer func() { execCommand = exec.Command }()
+
+	defer func() { execLookPath = exec.LookPath }()
+
+	execLookPath = func(name string) (string, error) {
+		return "", nil
+	}
+
+	out := isDomainJoined()
+	assert.False(t, out)
+}
+
+func TestExecHelperProcess(t *testing.T) {
+	if os.Getenv("GO_WANT_HELPER_PROCESS") != "1" {
+		return
+	}
+
+	mockRealmList := "contoso.com\n  type: kerberos\n  realm-name: CONTOSO.COM\n  domain-name: contoso.com\n  configured: kerberos-member\n  server-software: active-directory\n  client-software: sssd\n  required-package: oddjob\n  required-package: oddjob-mkhomedir\n  required-package: sssd\n  required-package: adcli\n  required-package: samba-common-tools\n  login-formats: %U@contoso.com\n  login-policy: allow-realm-logins"
+	fmt.Fprintf(os.Stdout, mockRealmList)
```

Comment:
nit: is this needed?

Reply:
this how we pass the response back if I understood other examples right